imbi/endpoints/project_ssm_configuration.py

```diff
@@ -0,0 +1,211 @@
+import os
+import re
+import typing
+import urllib.parse
+
+import aioredis
+import botocore.exceptions
+import sprockets.mixins.http
+from tornado import web
+
+from imbi import errors, oauth2, user
+from imbi.clients import aws
+from imbi.endpoints import base
+
+
+class CollectionRequestHandler(sprockets.mixins.http.HTTPClientMixin,
+                               base.ValidatingRequestHandler):
+    GET_PROJECT_INFO_SQL = re.sub(
+        r'\s+', ' ', """\
+        SELECT n.id AS namespace_id,
+               n.aws_ssm_slug AS namespace_slug,
+               p.slug AS project_slug,
+               t.slug AS project_type_slug,
+               p.configuration_type,
+               p.environments
+          FROM v1.namespaces AS n
+          JOIN v1.projects AS p
+            ON p.namespace_id = n.id
+          JOIN v1.project_types AS t
+            ON p.project_type_id = t.id
+         WHERE p.id = %(project_id)s""")
+
+    GET_ROLE_ARN_SQL = re.sub(
+        r'\s+', ' ', """\
+        SELECT role_arn
+          FROM v1.aws_roles
+         WHERE environment = %(environment)s
+           AND namespace_id = %(namespace_id)s""")
+
+    @staticmethod
+    def path_prefix(template: str, project_slug: str, project_type_slug: str,
+                    namespace_slug: str) -> str:
+        return template.replace('{project_slug}', project_slug).replace(
+            '{project_type_slug}',
+            project_type_slug).replace('{namespace_slug}', namespace_slug)
+
+    @property
+    def _redis(self) -> aioredis.Redis:
+        return self.application.session_redis
+
+    async def get(self, *args, **kwargs) -> None:
+        result = await self.postgres_execute(
+            self.GET_PROJECT_INFO_SQL, {'project_id': kwargs['project_id']},
+            'get-project-namespace-id')
+        project_info = result.row
+
+        if project_info['configuration_type'] != 'ssm':
+            raise web.HTTPError(404,
+                                reason='Wrong configuration type for project')
+
+        imbi_user = await self.get_current_user()
+        env = os.environ.get('ENVIRONMENT', 'production').lower()
+        google_integration = await oauth2.OAuth2Integration.by_name(
+            self.application, f'google-{env}')
+        tokens = await google_integration.get_user_tokens(imbi_user)
+        if len(tokens) == 0:
+            raise web.HTTPError(
+                401,
+                reason=
+                'Not authorized to access SSM in this namespace & environment')
+        google_tokens = tokens[0]
+
+        ssm_path_prefix = self.path_prefix(
+            self.application.settings['project_configuration']
+            ['ssm_prefix_template'], project_info['project_slug'],
+            project_info['project_type_slug'], project_info['namespace_slug'])
```

Comment:
I think that we have to ensure that there is a trailing slash. We could do that when we parse the configuration or in the `path_prefix()` helper.

Reply:
Just checked, `get_parameters_by_path` works both with and without a trailing `/`.